ports/stm32f4/common-hal/busio/UART.c

```diff
@@ -36,45 +36,455 @@
 #include "supervisor/shared/translate.h"
 
 #include "tick.h"
+#include "stm32f4xx_hal.h" 
+
+STATIC bool reserved_uart[MAX_UART];
```

Comment:
`MAX_UART` looks to be 6. Is it always the case that all the USART1, 2, etc. are defined, so there are no unset slots in `reserved_uart` after `uart_reset()` runs?

Reply:
Because ST often skips peripherals (IE including UARTS 1, 2, and 6, but not 3, 4 or 5), I always have the reserved_periph[] array be the maximum possible size and assume elements of it will not be used (always false). It does require the periph file to be set up correctly (all pins map to existing elements on `mcu_uart_banks[]` which also uses NULL elements) so I'll add a check for that, even if it shouldn't ever get hit.